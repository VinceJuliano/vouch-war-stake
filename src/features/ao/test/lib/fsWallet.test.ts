import { describe, test, expect } from "vitest";
import { createFsWallet, loadTestWallet } from "./fsWallet";

describe("createFsWallet", () => {
  test("loads jwk", async () => {
    const wallet = await createFsWallet("./fixtures/test_jwk.json")({
      permissionsRequested: [],
    });
    expect(wallet.success).toBe(true);
    expect(wallet).toMatchSnapshot();
  });
});

describe("loadTestWallet", () => {
  test("loads test wallet", async () => {
    const wallet = await loadTestWallet();
    expect(wallet).toMatchSnapshot();
  });
});
