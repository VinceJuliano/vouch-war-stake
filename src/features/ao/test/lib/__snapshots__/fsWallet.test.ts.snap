// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createFsWallet > loads jwk 1`] = `
{
  "result": {
    "address": "HHz5nNVEzkALjmZq8cdIdubMBJCpIFOvzO5uHfU5nQI",
    "anonymous": false,
    "signer": [Function],
    "type": "Keyfile",
  },
  "success": true,
}
`;

exports[`loadTestWallet > loads test wallet 1`] = `
{
  "address": "HHz5nNVEzkALjmZq8cdIdubMBJCpIFOvzO5uHfU5nQI",
  "anonymous": false,
  "signer": [Function],
  "type": "Keyfile",
}
`;
